### PR TITLE
Specifically search for the smartctl command and alert if it's missing

### PR DESCRIPTION
**Pull request by peternewman**
This also helps for if you're using e.g. passwordless sudo for a particular user, as you get a more obvious error message!

### CODE REVIEW

File: check_smart.pl [hunk 1]
```diff
@@ -181,8 +181,19 @@
...
+my $smart_command = undef;
+foreach my $path (@sys_path) {
+	if (-x "$path/smartctl") {
+		$smart_command = "sudo $path/smartctl";
+		last;
+	}
+}
+
+if (!defined($smart_command)) {
+	print "UNKNOWN - Could not find smartctl in " . join(", ", @sys_path) . "\n";
```
Comment (Napsty) [inline check_smart.pl:193]: What about `UNKNOWN - Could not find executable smartctl in` as output. 
For example, when smartctl is not executable for whatever reason:

```
ckadm@mintp /tmp $ whereis smartctl
smartctl: /usr/sbin/smartctl /usr/share/man/man8/smartctl.8.gz

ckadm@mintp /tmp $ sudo chmod 666 /usr/sbin/smartctl

ckadm@mintp /tmp $ ./check_smart.pl -g "/dev/sd*[a-z]" -i ata
UNKNOWN - Could not find smartctl in /usr/bin, /bin, /usr/sbin, /sbin, /usr/local/bin, /usr/local/sbin
```

Comment (peternewman) [reply check_smart.pl:193]: Sounds very sensible. Do you want to do it as a suggestion so you get the credit, or are you not too bothered?

Comment (Napsty) [reply check_smart.pl:193]: Not bothered at all

Comment (peternewman) [reply check_smart.pl]: Done.

**Review by Napsty — APPROVED** [review]

**Pull request merged**